addons/web/models/models.py

```diff
@@ -65,10 +65,14 @@ def web_search_read(self, domain=None, fields=None, offset=0, limit=None, order=
                 'length': 0,
                 'records': []
             }
-        if limit and (len(records) == limit or self.env.context.get('force_search_count')):
+        current_length = len(records) + (offset or 0.0)
```

Comment:
Why `or 0.0` ? The default value is `0` and the offset parameters should be always a int.